### PR TITLE
expose outstanding timer count

### PR DESCRIPTION
**Pull request by jameshfisher**
I want to expose this so we can add it as a metric. If we're near the limit, it's a big operational risk.

### CODE REVIEW

File: ext/eventmachine.h [hunk 1]
```diff
@@ -51,7 +51,8 @@ extern "C" {
 	bool evma_run_machine_once();
 	void evma_run_machine();
 	void evma_release_library();
-	const uintptr_t evma_install_oneshot_timer (int seconds);
+	const size_t evma_get_outstanding_timer_count (void);
+	const uintptr_t evma_install_oneshot_timer (uint64_t milliseconds);
```
Comment (BenPope) [inline ext/eventmachine.h]: Was this change intentional?

File: ext/cmain.cpp [hunk 2]
```diff
@@ -95,6 +95,15 @@ extern "C" void evma_run_machine()
 	EventMachine->Run();
 }
 
+/********************************
+evma_get_outstanding_timer_count
+********************************/
+
+extern "C" const size_t evma_get_outstanding_timer_count (void)
```
Comment (BenPope) [inline ext/cmain.cpp]: Since this is being built with a C++ compiler, the `(void)` can be `()`.

File: ext/em.cpp [hunk 3]
```diff
@@ -1136,6 +1136,14 @@ void EventMachine_t::_RunTimers()
 	}
 }
 
+/*********************************************
+EventMachine_t::_GetNumberOfOutstandingTimers
+*********************************************/
+
+size_t EventMachine_t::GetOutstandingTimerCount(void)
```
Comment (BenPope) [inline ext/em.cpp]: `()`

File: ext/em.h [hunk 4]
```diff
@@ -142,6 +142,7 @@ class EventMachine_t
 		void ScheduleHalt();
 		bool Stopping();
 		void SignalLoopBreaker();
+		size_t GetOutstandingTimerCount(void);
```
Comment (BenPope) [inline ext/em.h]: `()`

File: ext/eventmachine.h [hunk 5]
```diff
@@ -51,7 +51,8 @@ extern "C" {
 	bool evma_run_machine_once();
 	void evma_run_machine();
 	void evma_release_library();
-	const uintptr_t evma_install_oneshot_timer (int seconds);
+	const size_t evma_get_outstanding_timer_count (void);
```
Comment (BenPope) [inline ext/eventmachine.h]: `()`

File: ext/rubymain.cpp [hunk 6]
```diff
@@ -254,6 +254,14 @@ static VALUE t_run_machine (VALUE self UNUSED)
 	return Qnil;
 }
 
+/*****************************
+t_get_outstanding_timer_count
+*****************************/
+
+static VALUE t_get_outstanding_timer_count (void)
```
Comment (BenPope) [inline ext/rubymain.cpp]: `()`

File: ext/cmain.cpp (hunk 2 above)
Comment (jameshfisher) [reply ext/cmain.cpp]: Aha, I don't really know C++ 👍

File: ext/eventmachine.h (hunk 1 above)
Comment (jameshfisher) [reply ext/eventmachine.h]: No, thank you, this was a borked merge!

File: ext/eventmachine.h [hunk 7]
```diff
@@ -51,6 +51,7 @@ extern "C" {
 	bool evma_run_machine_once();
 	void evma_run_machine();
 	void evma_release_library();
+	const size_t evma_get_outstanding_timer_count (void);
```
Comment (BenPope) [inline ext/eventmachine.h]: Missed one!
